### PR TITLE
461 add support to load decathalon datalist

### LINKED ISSUE #461: Add utility function to load dataset from JSON file
**Is your feature request related to a problem? Please describe.**
Most of public datasets has a separate file to store `datalist`, let's add a JSON parser first based on the Decathalon dataset format first.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #461 .

### Description
As Decathalon challenge dataset is very rich and very popular in medical domain, many researchers and students use Decathalon dataset to learn medical DL skills, and we also have notebooks and examples based on Decathalon dataset.
So this PR added support to load Decathalon datalist from the JSON config file.
Users can also convert their own datalist to Decathalon format and use this tool.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/data/decathalon_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    Those dataset.json files
+
+    Args:
+        is_segmentation (str): whether the datalist is for segmentation task
+        data_list_key (str): the key to get a list of dictionary to be used
+        data_list_file_path (str): the path to the json file of datalist
+        base_dir (str): the base directory of the dataset, if None, use the datalist directory
+
+    Returns a list of data items, each of which is a dict keyed by element names
+
```
Comment (wyli) [inline monai/data/decathalon_dataloader.py:53]: could have an example of the returned result here
```
[{'image': '/workspace/data/chest_19.nii.gz',  'label': 0}, 
 {'image': '/workspace/data/chest_31.nii.gz',  'label': 1}]
```

File: monai/data/decathalon_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            raise ValueError("data item must be dict.")
+        for k, v in item.items():
+            if k == "image":
+                item[k] = _compute_path(base_dir, v)
+            elif is_segmentation and k == "label":
+                item[k] = _compute_path(base_dir, v)
+    return items
+
+
+def load_decathalon_datalist(is_segmentation, data_list_key, data_list_file_path, base_dir=None):
```
Comment (wyli) [inline monai/data/decathalon_dataloader.py:40]: shall we put some default value?
```python
load_decathalon_datalist(data_list_file_path, is_segmentation=True, data_list_key='training', base_dir=None):
```

File: monai/data/decathalon_dataloader.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/data/decathalon_dataloader.py:63]: updated in the latest commit.
Thanks.

File: monai/data/decathalon_dataloader.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/data/decathalon_dataloader.py]: updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review, I updated according to the comments.
Could you please help review it again?
Thanks.

File: monai/data/decathalon_dataloader.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    Json file is similar to what you get from http://medicaldecathlon.com/
+    Those dataset.json files
+
+    Args:
+        data_list_file_path (str): the path to the json file of datalist
+        is_segmentation (bool): whether the datalist is for segmentation task, default is True
+        data_list_key (str): the key to get a list of dictionary to be used, default is "training"
+        base_dir (str): the base directory of the dataset, if None, use the datalist directory
+
+    Returns a list of data items, each of which is a dict keyed by element names, for example:
```
Comment (wyli) [inline monai/data/decathalon_dataloader.py:52]: this is still broken in the docs, I've checked that the following works
```
    Returns a list of data items, each of which is a dict keyed by element names, for example:

    .. code-block::

        [
            {'image': '/workspace/data/chest_19.nii.gz',  'label': 0},
            {'image': '/workspace/data/chest_31.nii.gz',  'label': 1}
        ]

```

Comment (Nic-Ma) [reply monai/data/decathalon_dataloader.py:52]: Thanks for your reminder. Added .. code-block:: in the latest commit.

**Pull request merged**